### PR TITLE
Clerk SDK Spec

### PR DESCRIPTION
**Pull request by LekoArts**
> [!IMPORTANT]
> 🔎 Previews:
>
> - https://clerk.com/docs/pr/1344/references/sdk/overview

Adds a SDK Development spec.

Fixes ECO-102

### CODE REVIEW

File: docs/references/sdk/philosophy.mdx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+For your SDK, follow the framework's or language's best practices and consider using their specialized patterns over generalized ones (e.g. special script loading). For your users, provide APIs to leverage those best practices.
+
+## Documentation
+
+Spend time writing an exhaustive, yet organized `README` that enables your users to get started with your SDK. If you want to go above and beyond, create your own documentation site (recommended: [Starlight](https://starlight.astro.build/)). Provide examples on how to use your SDK, e.g. with a separate example repository.
+
+## Convenience
+
+Make SDK usage convenient. If the SDK can handle something under the hood which otherwise would require an action by the user, do it. Clerk's components are its API as that abstraction is powerful and convenient.
```
Comment (alexisintech) [inline docs/references/sdk/philosophy.mdx:28]: "Clerk's components are its API" doesn't make sense - what were you trying to say here?

File: docs/references/sdk/philosophy.mdx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+Ensure compatibility with the upstream framework or language. Have reasonable backward compatibility.
+
+## API surface
+
+Keep the API surface small. Most often you’ll only need to expose the options for the underlying Clerk SDKs and pass those through.
+
+## Testing
+
+Add End-to-End tests to regularly test hot paths of your SDK. Functionality like sign-in, sign-up, and other UI components have high priority. Inside the [testing guide](https://clerk.com/docs/testing/overview) you'll learn how to use testing tokens and which testing utilities are available.
```
Comment (alexisintech) [inline docs/references/sdk/philosophy.mdx:40]:
```suggestion
Add End-to-End tests to regularly test hot paths of your SDK. Functionality like sign-in, sign-up, and other UI components have high priority. Checkout the [testing guide](https://clerk.com/docs/testing/overview) to learn which testing utilities are available and how to use Clerk's Testing Tokens to bypass bot detection.
```

File: docs/manifest.json [hunk 3]
```diff
@@ -2570,6 +2570,39 @@
               "tag": "Community",
               "href": "https://github.com/anagstef/ngx-clerk?tab=readme-ov-file#ngx-clerk",
               "icon": "angular"
+            },
+            {
+              "title": "SDK Development",
```
Comment (LekoArts) [inline docs/manifest.json:2575]:
```suggestion
              "title": "SDK development",
```

File: docs/references/sdk/philosophy.mdx (hunk 1 above)
Comment (LekoArts) [reply docs/references/sdk/philosophy.mdx:30]: You might interact/use a product through a REST API or through an SDK that exposes certain low-level methods. That's the API surface that they expose to their users.

Clerk's API are the prebuilt React components. Just place them in your React application and you're good. It's super convenient.

That's what I wanted to convey 😊

File: docs/references/sdk/types.mdx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+- Backend-only
+- Fullstack
+
+Take a look at your framework and check which type it enables. Then place your SDK into one of these three categories when first creating it. Arguably, creating a fullstack SDK is the most work so feel free to start off with frontend-only/backend-only SDK — you can still make it fullstack at a later point. Also, the framework you’re building the SDK for might evolve in the future, so maybe naturally your SDK will also need to change over time.
+
+| SDK Type | FAPI HTTP client | BAPI HTTP client | Hotloading ClerkJS | `window.Clerk` UI components | Authorization header verification | `__session` cookie verification | Handshake support |
+| - | - | - | - | - | - | - | - |
+| Frontend-only | ✅ | ⬜ | ✅ | ✅ | ⬜ | ⬜ | ⬜ |
+| Backend-only | ⬜ | ✅ | ⬜ | ⬜ | ✅ | ✅ | ✅ |
+| Fullstack | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
```
Comment (panteliselef) [inline docs/references/sdk/types.mdx:20]: 🙃 This grey emoji, makes me think that something did not render correctly, may we switch to something else ?

File: docs/references/sdk/types.mdx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+## Frontend-only
+
+Frontend-only SDKs are used in frameworks that only support client-side rendering (CSR) or static site generation (SSG) with rehydration. The app is rendered in the browser using JavaScript and this bundle doesn’t contain any authenticated information itself but uses client-side code to fetch the authenticated data.
+
+**Examples:** Vite React, Vue, iOS/Swift
+
+**SDK Examples:** `@clerk/clerk-react`
```
Comment (panteliselef) [inline docs/references/sdk/types.mdx:28]: 🙃 Could this be link to the repo directory for our react package ?

**Review by panteliselef — APPROVED** [review]
Great Work !!

**Pull request merged**